python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -1280,6 +1280,45 @@ def update_replace_dict_and_conflicts(
         merged_attrs = merge_dicts(self.get_attributes_dict(), replaced_attributes)
         return self.__class__.dagster_internal_init(**merged_attrs)
 
+    def with_replaced_asset_specs(
+        self,
+        specs: Sequence[AssetSpec],
+        new_asset_keys_by_old_asset_key: Mapping[AssetKey, AssetKey],
+    ) -> "AssetsDefinition":
+        return self.__class__.dagster_internal_init(
+            **{
+                **self.get_attributes_dict(),
+                "specs": specs,
```

Comment:
One thing that is confusing here is why the dep swapping is not done in *this* layer.  This function is really only useful in the context of its single caller as it will (misleadingly?) not remap the deps even though it receives a mapping dictionary.